test/unit/test_keyring_multi.py

```diff
@@ -0,0 +1,242 @@
+    assert len(test.encrypted_data_keys) == len(get_encryption_materials_without_data_key().encrypted_data_keys)
+
+    test = test_multi_keyring.on_encrypt(encryption_materials=get_encryption_materials_with_encrypted_data_key())
+    assert len(test.encrypted_data_keys) == len(get_encryption_materials_with_encrypted_data_key().encrypted_data_keys)
+
+
+def test_on_decrypt_when_data_encryption_key_given():
+    test_multi_keyring = MultiKeyring(generator=mock_generator, children=[mock_child_1, mock_child_2])
+    test_multi_keyring.on_decrypt(decryption_materials=get_decryption_materials_with_data_key(), encrypted_data_keys=[])
+    for keyring in test_multi_keyring._decryption_keyrings:
+        keyring.on_decrypt.assert_not_called()
+
+
+def test_on_decrypt_every_keyring_called_when_data_encryption_key_not_added():
+    mock_generator.on_decrypt.return_value = get_decryption_materials_without_data_key()
+    mock_child_1.on_decrypt.return_value = get_decryption_materials_without_data_key()
+    mock_child_2.on_decrypt.return_value = get_decryption_materials_without_data_key()
+
+    test_multi_keyring = MultiKeyring(generator=mock_generator, children=[mock_child_1, mock_child_2])
+    test_multi_keyring.on_decrypt(
+        decryption_materials=get_decryption_materials_without_data_key(), encrypted_data_keys=[]
+    )
+
+    for keyring in test_multi_keyring._decryption_keyrings:
+        keyring.on_decrypt.assert_called()
+
+
+# def test_no_keyring_called_after_data_encryption_key_added_when_data_encryption_key_not_given():
+#     mock_generator.on_decrypt(decryption_materials=get_decryption_materials_without_data_key(),
+#                               encrypted_data_keys=[])
+#     mock_generator.on_decrypt.return_value = get_decryption_materials_without_data_key()
+#
+#     mock_child_3.on_decrypt(decryption_materials=mock_generator.on_decrypt.return_value,
+#                             encrypted_data_keys=[])
+#     mock_child_3.on_decrypt.return_value = get_decryption_materials_with_data_key()
+#
+#     mock_child_1.on_decrypt(decryption_materials=mock_child_3.on_decrypt.return_value,
+#                             encrypted_data_keys=[])
+#     mock_child_1.on_decrypt.return_value = get_decryption_materials_with_data_key()
+#
+#     mock_child_2.on_decrypt(decryption_materials=mock_child_1.on_decrypt.return_value,
+#                             encrypted_data_keys=[])
+#     mock_child_2.on_decrypt.return_value = get_decryption_materials_with_data_key()
+#
+#     test_multi_keyring = MultiKeyring(
+#         generator=mock_generator,
+#         children=[
+#             mock_child_3,
+#             mock_child_1,
+#             mock_child_2,
+#         ]
+#     )
+#     test_multi_keyring.on_decrypt(decryption_materials=get_decryption_materials_without_data_key(),
+#                                   encrypted_data_keys=[])
+#     mock_generator.on_decrypt.assert_called()
+#     mock_child_3.on_decrypt.assert_called()
```

Comment:
Similar to the issue with `assert_not_called`, this was only added in Python 3.6 :(

Use `assert X.called` instead.

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called